cpp/src/arrow/filesystem/azurefs.cc

```diff
@@ -147,6 +159,10 @@ Status AzureOptions::ExtractFromUriQuery(const Uri& uri) {
     } else if (kv.first == "background_writes") {
       ARROW_ASSIGN_OR_RAISE(background_writes,
                             ::arrow::internal::ParseBoolean(kv.second));
+    } else if (std::find(sas_token_query_parameters.begin(),
```

Comment:
Can we use `std::binary_search()` with sorted `sas_token_query_parameters`?

https://en.cppreference.com/w/cpp/algorithm/binary_search

Reply:
Could do but I would be a bit concerned about keeping `sas_token_query_parameters` sorted. It looks like C++20 allows using `std::sort` with `constexpr` but I beleive arrow currently uses C++17. If you are concerned about the complexity of the lookup I think my preference would be to use a `std::map` and forget about trying to make it a `constexpr`.

Reply:
`std::set` is OK.